### PR TITLE
You can('t) whisper into comms in crit.

### LINKED ISSUE #54878: You can whisper into comms in crit.
Caused by https://github.com/tgstation/tgstation/pull/53117?

Dunno if this is intended or not.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Fixing radios and some saymodes not properly respecting stats.

## Why It's Good For The Game
This will close #54878. 

## Changelog
:cl: 
fix: Spessmen can't whisper into comms in crit anymore (excluding handheld radios and intercomms).
fix: Humans with binary encrypted headsets shouldn't be able to speak into the binary channel when unconscious/critted anymore.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
This isn't the right way to go about this, neither were the last 2 attempts tho, so you're on par here.

The issue here is mainly that being in softcrit should only allow you to use certain message modifiers, or blacklist a few. The way it currently works is just toggling whispering on at all times, due to #53117.

My original pr was broken for semi similar reasons, if any of your message mods were alright you passed, and the code for that was broken too. 

On a totally separate note, aliens talking in crit won't work properly because aliens never get a message mod or radio extension (Code for most things that use :{something), including admin chat, and left/right hand radio talking) assigned to them, instead they are dealt with using saymodes based off the radio key. So the check for aliens in that block will never fire.

What you should do is make a blacklist of values that the assoc list message mods can have, and if any exist when the status checking block is reached, remove them from the list.

**Comment by Ghommie** [comment]
Done more or less as you suggested. The xeno hivemind is supposed to work even if critted or unconscious.

**Comment by BootlegBow** [comment]
Slimelink is also supposed to work when critted or unconscious, keep that in mind

**Comment by Ghommie** [comment]
Slimelink currently doesn't rely on say code right now. It should be easy as pie to whitelist if ever refactored though.

**Comment by LemonInTheDark** [comment]
alllright then pending

File: code/modules/mob/living/living_say.dm [hunk 1]
```diff
@@ -26,7 +26,6 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 	// Misc
 	RADIO_KEY_AI_PRIVATE = RADIO_CHANNEL_AI_PRIVATE, // AI Upload channel
-	MODE_KEY_VOCALCORDS = MODE_VOCALCORDS,		// vocal cords, used by Voice of God
```
Comment (AnturK) [inline code/modules/mob/living/living_say.dm:29]: Was this mode/key unused ?

Comment (LemonInTheDark) [reply code/modules/mob/living/living_say.dm:29]: clerical error I assume?

Comment (Ghommie) [reply code/modules/mob/living/living_say.dm:29]: Same reason why there is no `MODE_KEY_ALIEN` or why `MODE_KEY_BINARY` isn't on the list, it's a saymode key and doesn't rely on `GLOB.department_radio_keys`.

**Pull request merged**